pkg/kv/dist_sender.go

```diff
@@ -900,23 +906,32 @@ func (ds *DistSender) sendPartialBatch(
 	}
 	var reply *roachpb.BatchResponse
 	var pErr *roachpb.Error
+	var err error
+	var intersected roachpb.RSpan
+	var truncBA roachpb.BatchRequest
+	var positions []int
 
 	isReverse := ba.IsReverse()
 
-	// Truncate the request to range descriptor.
-	intersected, err := rs.Intersect(desc)
-	if err != nil {
-		return response{pErr: roachpb.NewError(err)}
-	}
-	truncBA, positions, err := truncate(ba, intersected)
-	if len(positions) == 0 && err == nil {
-		// This shouldn't happen in the wild, but some tests exercise it.
-		return response{
-			pErr: roachpb.NewErrorf("truncation resulted in empty batch on %s: %s", intersected, ba),
+	if needsTruncate {
+		// Truncate the request to range descriptor.
+		intersected, err = rs.Intersect(desc)
```

Comment:
can this overwrite `rs` instead of assigning to `intersected`? seems like it'd reduce the possibility of doing the wrong thing. `ba` and `truncBA` seem to demonstrate the same oddity.